lib/osf-components/addon/components/storage-provider-manager/osf-storage-manager/component.ts

```diff
@@ -0,0 +1,100 @@
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import { waitFor } from '@ember/test-waiters';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { restartableTask, task } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import FileProviderModel from 'ember-osf-web/models/file-provider';
+import NodeModel from 'ember-osf-web/models/node';
+import { FileSortKey } from 'ember-osf-web/packages/files/file';
+import OsfStorageFile from 'ember-osf-web/packages/files/osf-storage-file';
+
+interface Args {
+    target: NodeModel;
+}
+
+export default class OsfStorageManager extends Component<Args> {
+    @tracked storageProvider?: FileProviderModel;
+    @tracked rootFolder?: OsfStorageFile;
+    @tracked currentFolder?: OsfStorageFile;
+    @tracked displayItems: OsfStorageFile[] = [];
+    @tracked filter = '';
+    @tracked sort = FileSortKey.AscDateModified;
+    @tracked currentPage = 1;
+
+    constructor(owner: unknown, args: Args) {
+        super(owner, args);
+        assert('@target must be provided', this.args.target);
+        taskFor(this.getRootFolderItems).perform();
+    }
+
+    @restartableTask
+    @waitFor
+    async getRootFolderItems() {
+        await taskFor(this.getRootFolder).perform();
+        await taskFor(this.getCurrentFolderItems).perform();
+    }
+
+    @restartableTask
+    @waitFor
+    async getRootFolder() {
+        if (this.args.target) {
+            const fileProviders = await this.args.target.files;
+            this.storageProvider = fileProviders.findBy('name', 'osfstorage') as FileProviderModel;
+            this.rootFolder = new OsfStorageFile(await this.storageProvider.rootFolder);
+            this.currentFolder = this.rootFolder;
+        }
+    }
+
+    @restartableTask
+    @waitFor
+    async getCurrentFolderItems() {
+        if (this.currentFolder) {
+            const items = await this.currentFolder.getFolderItems(this.currentPage, this.sort, this.filter);
+            const newArray = [...this.displayItems];
+            newArray.push(...items);
+            this.displayItems = newArray;
```

Comment:
I think [notifyPropertyChange](https://api.emberjs.com/ember/3.26/classes/Observable/methods/notifyPropertyChange?anchor=notifyPropertyChange) would make more sense to future code maintainers.

Reply:
It seems there isn't a `notifyPropertyChange` hook on the component itself. All of the examples in our code base uses the hook as a class method of an ember-data model, which is probably an `Observable`.

Reply:
I thought it was a function on the array itself, that gets put on when it becomes tracked. But I may be misremembering.

Reply:
No, I was misremembering. How about:

```js
import { notifyPropertyChange } from '@ember/object';

export default class MyComponent extends Component {
  @action addListItem() {
    this.items.push(randomCatUrl());
    notifyPropertyChange(this, 'items');
  }
}
```